src/apps/HeyGPT/HeyGPT/Services/ActivationService.cs

```diff
@@ -1,60 +1,168 @@
-﻿using HeyGPT.App.Activation;
+﻿using System.Diagnostics;
+using HeyGPT.App.Activation;
 using HeyGPT.App.Contracts.Services;
+using HeyGPT.App.Helpers;
 using HeyGPT.App.Views;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.UI.Xaml;
-using Microsoft.UI.Xaml.Controls;
+using Microsoft.Windows.AppLifecycle;
 
 namespace HeyGPT.App.Services;
 
+/// <summary>
+/// Manages application activation including initial application launch
+/// and advanced activation scenarios such as handling URL protocols.
+/// </summary>
 public class ActivationService : IActivationService
 {
-    private readonly ActivationHandler<LaunchActivatedEventArgs> _defaultHandler;
+    private readonly IActivationHandler _defaultHandler;
     private readonly IEnumerable<IActivationHandler> _activationHandlers;
     private readonly IThemeSelectorService _themeSelectorService;
-    private UIElement? _shell = null;
+    private UIElement? _shell;
+    private int _activationCount;
 
-    public ActivationService(ActivationHandler<LaunchActivatedEventArgs> defaultHandler, IEnumerable<IActivationHandler> activationHandlers, IThemeSelectorService themeSelectorService)
+    //public ActivationService(ActivationHandler<AppActivationArguments> defaultHandler, IEnumerable<IActivationHandler> activationHandlers, IThemeSelectorService themeSelectorService)
+    public ActivationService(
+        [FromKeyedServices(ActivationHandlerKeys.DefaultServiceKey)] IActivationHandler defaultHandler,
+        [FromKeyedServices(ActivationHandlerKeys.OtherServiceKey)] IEnumerable<IActivationHandler> activationHandlers,
+        IThemeSelectorService themeSelectorService)
     {
         _defaultHandler = defaultHandler;
         _activationHandlers = activationHandlers;
         _themeSelectorService = themeSelectorService;
```

Comment:
We injected the default handler and a list of Other activation handlers.   Still unsure about the default handler idea... Need to look more at the navigation service.